### DISCUSSION TITLE
Search results isn't pulling in all documents

### DISCUSSION THREAD

**Discussion by nicolesherman00** [microsoft-search/pnp-modern-search#2537] [Q&A]
Hello!!
The search results web parts seem to not be bringing in all of our items. 
I have been told that the total amounts on the PNP Modern Search webparts are not totally accurate but mine are showing miles apart and documents are missing from search.
As you can see:
1. Is the query I am using pointing to a document library - You might see a discrepancy in the library names from whats in the query to what is in a further screenshot but the library was renamed.
2. the count that is being pulled (again, I do realize its not totally accurate)
![image](https://user-images.githubusercontent.com/22782498/201422533-2e615b3f-6cc1-4254-b3a6-d38c01a2c4be.png)
Here is the amount of items in the CHAS Job Aids library
![image](https://user-images.githubusercontent.com/22782498/201422866-22f322b7-8dc8-4045-aeb2-bdd02d34230e.png)
Now if we go into the CHAS Job aids library, we have sub folders.. within each subfolder all items within have a refinablestring assigned column we are using to filter by. So for this example, we are going to navigate to the Behavioral heath FOLDER, where all items within have the Behavioral Health choice added to that column..  But as we filter by behavioral health from the PNP modern search webparts, not all items are showing up.
60 items when I select all documents in this folder
![image](https://user-images.githubusercontent.com/22782498/201424379-55aea967-e9be-47ab-b665-42895fafd2f1.png)
30 items when I select the Behavioral Health folder - I have confirmed all documents have the choice tag applied and have even removed and re-added the tag but that his not help.
![image](https://user-images.githubusercontent.com/22782498/201424678-8c0c7c64-912c-48d8-8455-41de4bb28364.png)
When we go into the pnp modern search bar and type 'bh special' it only brings up one item when multiple should be there!
RESULTS
![image](https://user-images.githubusercontent.com/22782498/201425216-3c57d30f-fbaa-4a8c-bc5b-18e10bf948e9.png)
HOW THE LIBRARY LOOKS
![image](https://user-images.githubusercontent.com/22782498/201425254-29a6507c-4c97-4d43-810a-ca62dbe806f8.png)

Any thoughts on what could be the issue? Am I missing anything you need to know?

**Comment by wobba** [comment]
If you want to see if it brings all item, don't look at the "total" count which is an estimate, but page thru the results, which I believe works up to 500 items. For an end-user they hardly page this far anyways. Using "count" and "paging" to validate that all items is not the way to go. Then you should rather use powershell (`Submit-PnPSearchQuery -All`) or similar and page based on docid (https://learn.microsoft.com/en-us/sharepoint/dev/general-development/pagination-for-large-result-sets) .

Might be better if you turn off the result count in the UX if it distracts. Or you can start a support case with Microsoft to raise your concern/scenario.

**Comment by Mareq80** [comment]
I had that recently and for some reason the indexer didn´t update field value in the index. You can check for that. 
1. Enable the debug display template. 
2. Do a search for all items using *
3. Find the item which should show up in filtered query. Review the field that is used in filter does it contain value you are looking for or is it null? 

In my case the values upon which filter should work were not part of search index (but saved in the list). Search is using index not making direct queries to the list. In my case resolution was simple I just had to trigger the re-indexing of the site (i made some changes to the manged properties so re indexing just the library would´t suffice)
